Setting securityContext supplementalGroups causes Stackoverflow error

**Describe the bug**
After setting a supplementalGroup in the securityContext of the pod template for zookeeper and kafka the strimzi-cluster-operator runs into a Stackoverflow error and does not create any resources (e.g. the zookeeper statefulset).

For the corresponding field in the kafka CRD see https://github.com/strimzi/strimzi-kafka-operator/blob/32f8c7f22bf4bea61cbf2d3f58c8da56c4aa02d3/install/cluster-operator/040-Crd-kafka.yaml#L2283

**To Reproduce**
Steps to reproduce the behavior:
1. Set up the strimzi operator in version 0.22.1 within a namespace in OpenShift 4.5 with OperatorHub
2. Create Custom Resource with a supplementalGroup for zookeeper and kafka (template->pod->securityContext->supplementalGroups...), see minimal yaml sample below
3. See error in strimzi operator pod log -> strimzi operator does not create any resources/pods for the zookeeper/kafka resources and logs a Stackoverflow exception (see log snippet below).

**Expected behavior**
StatefulSet for zookeeper is created with pods having the given supplementalGroup(s) set.

**Environment:**
 - Strimzi version: 0.22.1
 - Installation method: OperatorHub.io
 - Kubernetes cluster: OpenShift 4.5.38
 - Infrastructure: on premise

**YAML files and logs**
YAML for a rather minimal kafka resource sample containing supplementalGroups to reproduce:
```
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: minimal-kafka-cluster
spec:
  entityOperator:
    topicOperator: {}
    userOperator: {}
  kafka:
    listeners:
    - name: plain
      port: 9092
      type: internal
      tls: false
    - name: tls
      port: 9093
      type: internal
      tls: true
    livenessProbe:
      failureThreshold: 10
      initialDelaySeconds: 600
      periodSeconds: 15
      timeoutSeconds: 5
    readinessProbe:
      failureThreshold: 10
      initialDelaySeconds: 90
      periodSeconds: 15
      timeoutSeconds: 5
    replicas: 3
    resources:
      limits:
        memory: 4Gi
      requests:
        memory: 2Gi
    storage:
      deleteClaim: false
      size: 1Gi
      type: persistent-claim
    template:
      pod:
        securityContext:
          supplementalGroups:
            - 971001234
  zookeeper:
    replicas: 3
    resources:
      limits:
        memory: 1Gi
      requests:
        memory: 512Mi
    storage:
      deleteClaim: false
      size: 1Gi
      type: persistent-claim
    template:
      pod:
        securityContext:
          supplementalGroups:
            - 971001234
```

Log of strimzi-cluster-operator pod after applying the above kafka resource:
```
2021-05-25 11:13:07 ERROR AbstractOperator:410 - Reconciliation #51(watch) Kafka(***/minimal-kafka-cluster): Reconciliation failed
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:80) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.AbstractOperator.validate(AbstractOperator.java:433) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.AbstractOperator.lambda$reconcile$7(AbstractOperator.java:252) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.AbstractOperator.lambda$withLock$14(AbstractOperator.java:397) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.vertx.core.shareddata.impl.LocalAsyncLocks$LockWaiter.lambda$acquireLock$1(LocalAsyncLocks.java:65) ~[io.vertx.vertx-core-3.9.1.jar:3.9.1]
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366) ~[io.vertx.vertx-core-3.9.1.jar:3.9.1]
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38) ~[io.vertx.vertx-core-3.9.1.jar:3.9.1]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) [io.netty.netty-transport-4.1.60.Final.jar:4.1.60.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.Final]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:109) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:111) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:111) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:111) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at ... (stacktrace repeats the last two lines)
Caused by: java.lang.StackOverflowError
	at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at java.lang.Class.getClasses(Class.java:1752) ~[?:?]
	at jdk.internal.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:109) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:111) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visit(ResourceVisitor.java:111) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at io.strimzi.operator.common.model.ResourceVisitor.visitProperty(ResourceVisitor.java:159) ~[io.strimzi.operator-common-0.22.1.jar:0.22.1]
	at ... (stacktrace repeats the last two lines)
```

--- a/operator-common/src/main/java/io/strimzi/operator/common/model/ResourceVisitor.java
+++ b/operator-common/src/main/java/io/strimzi/operator/common/model/ResourceVisitor.java
@@ -147,7 +147,9 @@ public class ResourceVisitor {
             } else if (Collection.class.isAssignableFrom(returnType)) {
                 path.add(propertyName);
                 for (Object element : (Collection<?>) propertyValue) {
-                    if (element != null && !element.getClass().isEnum()) {
+                    if (element != null
+                            && !element.getClass().isEnum()
+                            && !isScalar(element.getClass())) {
                         visit(path, element, visitor);
                     }
                 }

--- a/operator-common/src/test/resources/example.yaml
+++ b/operator-common/src/test/resources/example.yaml
@@ -27,6 +27,11 @@ spec:
     storage:
       type: ephemeral
     version: 2.8.0
+    template:
+      pod:
+        securityContext:
+          supplementalGroups:
+            - 971001234
     tlsSidecar: {}
   topicOperator: {}
   zookeeper:

